tests/components/vizio/test_media_player.py

```diff
@@ -193,11 +195,20 @@
                 list_to_test.remove(app_to_remove)
 
         assert attr["source_list"] == list_to_test
-        assert app in attr["source_list"] or app == UNKNOWN_APP
-        if app == UNKNOWN_APP:
-            assert attr["source"] == ADDITIONAL_APP_CONFIG["name"]
-        else:
+
+        if app:
+            assert app in attr["source_list"] or app == UNKNOWN_APP
             assert attr["source"] == app
+            assert attr["app_name"] == app
+            if app == UNKNOWN_APP:
+                assert attr["app_id"] == app_config
+            else:
+                assert "app_id" not in attr
+        else:
+            assert attr["source"] == "CAST"
+            assert "app_id" not in attr
+            assert "app_name" not in attr
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5245099)